src/makePersistable.ts

```diff
@@ -0,0 +1,22 @@
+import { PersistenceStorageOptions, ReactionOptions } from './types';
+import { StorePersist } from './StorePersist';
+import { StorageConfiguration } from './StorageConfiguration';
+import { duplicatedStoreWarningIf } from './utils';
+
+export const makePersistable = <T extends { [key: string]: any }, P extends keyof T>(
+  target: T,
+  storageOptions: PersistenceStorageOptions<P>,
+  reactionOptions?: ReactionOptions
+): StorePersist<T, P> => {
+  const mobxPersistStore = new StorePersist(target, storageOptions, reactionOptions);
+
+  const hasPersistedStoreAlready = Array.from(StorageConfiguration.values())
+    .map((item) => item.storageName)
+    .includes(mobxPersistStore.storageName);
+
+  duplicatedStoreWarningIf(hasPersistedStoreAlready, mobxPersistStore.storageName);
+
+  StorageConfiguration.set(target, mobxPersistStore);
+
+  return mobxPersistStore;
+};
```

Comment:
@quarrant in reference to your comment:
> like get(target.storageName)

I don't think we can use the storage name like this:
```ts
// Usage Example
// makePersistable(this, { name: 'ProjectPageStore', properties: ['something'] });

// Changing StorageConfiguration.set to use storage name
StorageConfiguration.set(storageOptions.name, mobxPersistStore);
```


```ts
// Current Usage
clearPersistedStore(this);

// The user would need to pass in the name of the storage instead of `this`
clearPersistedStore('ProjectPageStore');

export const clearPersistedStore = async <T>(storageName: string): Promise<void> => {
  await StorageConfiguration.get(storageName)?.clearPersistedStore();
};
```

It seems we need to keep a reference to `this` to be able to have the user do `clearPersistedStore(this)` I don't see different way of doing it. I think your old kept a reference to the store also.

Reply:
```ts
// it's correctly
clearPersistedStore(this); 
```

i thought about something like this

```ts
export const clearPersistedStore = async <T>(target: T): Promise<void> => {
  await StorageConfiguration.get(target.storageName)?.clearPersistedStore();
};
```

Reply:
Can you fork this branch and get that working? I am not 100% how that would work.

Reply:
Ok. I'll try